pkg/cosign/env/env.go

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package env
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+type Variable string
+
+type VariableOpts struct {
+	Description string
+	Expects     string
+	Sensitive   bool
+}
+
+func (v Variable) String() string {
+	return string(v)
+}
+
+const (
+	VariableExperimental     Variable = "COSIGN_EXPERIMENTAL"
+	VariableDockerMediaTypes Variable = "COSIGN_DOCKER_MEDIA_TYPES"
+	VariablePassword         Variable = "COSIGN_PASSWORD"
+	VariablePKCS11Pin        Variable = "COSIGN_PKCS11_PIN"
+	VariablePKCS11ModulePath Variable = "COSIGN_PKCS11_MODULE_PATH"
+	VariableRepository       Variable = "COSIGN_REPOSITORY"
+)
+
+var (
+	environmentVariables = map[Variable]VariableOpts{
```

Comment:
One thing I'm noticing: this design centralizes the environment variables.

You could also consider a registry, where various parts of Cosign would register their own environment variables. (Closer to where they're used). I think *many* of these are used *a lot of places*, so that approach may not be great :shrug: 

I think this is fine as-is, I just want to make a note so that this decision is explicit.